documentation/solution-parser.md

```diff
@@ -0,0 +1,91 @@
+# Solution Parser
+
+## Expander
+
+`Expander` class is used to expand value from an expression, which can
+be function invocation. These expression (or embedded properties)
+have form - "$(property)".
+
+`Expander` handles intrinsic functions, which is a list of built-in functions
+(bunch of ifs).
```

Comment:
```suggestion
`Expander` handles intrinsic functions, which is a list of built-in functions.
```

It _is_ implemented as a bunch of ifs, but I would love for there to be some beautiful way to make that not true, and I don't think that's important for documentation.